sql/src/main/java/io/crate/executor/transport/TransportShardAction.java

```diff
@@ -78,27 +79,32 @@ protected boolean resolveIndex() {
     }
 
     @Override
-    protected WriteResult<ShardResponse> onPrimaryShard(Request shardRequest, IndexShard indexShard) throws Exception {
-        KillableWrapper<WriteResult<ShardResponse>> callable = new KillableWrapper<WriteResult<ShardResponse>>() {
-            @Override
-            public WriteResult<ShardResponse> call() throws Exception {
-                return processRequestItems(shardRequest.shardId(), shardRequest, killed);
-            }
-        };
+    protected WritePrimaryResult<Request, ShardResponse> shardOperationOnPrimary(Request shardRequest,
+                                                                                 IndexShard primary) throws Exception {
+        KillableWrapper<WritePrimaryResult<Request, ShardResponse>> callable =
+            new KillableWrapper<WritePrimaryResult<Request, ShardResponse>>() {
+                @Override
+                public WritePrimaryResult<Request, ShardResponse> call() throws Exception {
+                    Tuple<ShardResponse, Translog.Location> responseLocationTuple =
+                        processRequestItems(shardRequest.shardId(), shardRequest, killed);
```

Comment:
I'd prefer returning the WritePrimaryResult right here instead of using an intermediate tuple